packages/components/switch/src/SwitchInput.styles.ts

```diff
@@ -49,13 +49,25 @@ export const styles = cva(
 
 export type StylesProps = VariantProps<typeof styles>
 
+export const thumbWrapperStyles = cva(
+  [
+    'pointer-events-none absolute inset-none flex items-center',
+    'transition-all duration-200 ease-in-out',
+  ],
+  {
+    variants: {
+      checked: {
+        true: 'translate-x-full',
+        false: 'translate-x-none',
+      },
+    },
+  }
+)
+
 export const thumbStyles = cva(
   [
-    'absolute flex items-center justify-center',
+    'absolute left-none top-none flex items-center justify-center',
     'bg-surface',
-    'group-spark-state-checked:left-full group-spark-state-checked:-translate-x-full',
```

Comment:
The perf improvement comes from here, as explained in [this article](https://codeburst.io/low-performance-css-transitions-and-some-alternatives-f84ff35bf07) shared by Marius.

Layout properties such as `left` have bad performance when animated. The trick is to only use translateX if possible.